toolchain/check/check.cpp

```diff
@@ -1047,17 +1048,31 @@ auto CheckParseTrees(const SemIR::File& builtin_ir,
   llvm::SmallVector<UnitInfo*> ready_to_check;
   ready_to_check.reserve(units.size());
   for (auto& unit_info : unit_infos) {
-    if (const auto& packaging =
-            unit_info.unit->parse_tree->packaging_directive()) {
-      if (packaging->api_or_impl == Parse::Tree::ApiOrImpl::Impl) {
-        // An `impl` has an implicit import of its `api`.
-        auto implicit_names = packaging->names;
-        implicit_names.package_id = IdentifierId::Invalid;
-        TrackImport(api_map, nullptr, unit_info, implicit_names);
-      }
+    const auto& packaging = unit_info.unit->parse_tree->packaging_directive();
+    if (packaging && packaging->api_or_impl == Parse::Tree::ApiOrImpl::Impl) {
+      // An `impl` has an implicit import of its `api`.
+      auto implicit_names = packaging->names;
+      implicit_names.package_id = IdentifierId::Invalid;
+      TrackImport(api_map, nullptr, unit_info, implicit_names);
     }
 
     llvm::DenseMap<ImportKey, Parse::NodeId> explicit_import_map;
+
+    // Add the prelude import. It's added to explicit_import_map so that it can
+    // conflict with an explicit import of the prelude.
+    // TODO: Add --disable-prelude-import for `/no_prelude/` subdirs.
```

Comment:
Presumably this is just for tests?
```suggestion
    // TODO: Add --disable-prelude-import for `/no_prelude/` subdirs in testdata.
```

Reply:
Done